### PR TITLE
refine: refactor bucket migrate code

### PR DESCRIPTION
**Pull request by jingjunLi**
### Description

refine bucket migrate code

### Rationale

* Rename certain variables.
* Utilize produceBucketMigrateExecutePlan for both event handling and database loading.
* Abstract SPConflictChecker for generating bucketMigrateUnits, applicable during event handling and database loading.

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

File: modular/manager/bucket_migrate_scheduler.go [hunk 1]
```diff
@@ -69,24 +69,24 @@ func (f *PickDestGVGFilter) CheckGVG(gvgMeta *vgmgr.GlobalVirtualGroupMeta) bool
 // CheckGVGMetaConsistent verifies whether expectedSPGVGList completely matches with the migrateGVGUnitMeta loaded from the database.
 func CheckGVGMetaConsistent(chainMetaList []*virtualgrouptypes.GlobalVirtualGroup, dbMetaList []*spdb.MigrateGVGUnitMeta) bool {
 	if len(chainMetaList) == len(dbMetaList) {
-		chainGvgMaps := make(map[uint32]*virtualgrouptypes.GlobalVirtualGroup)
-		dbGvgMaps := make(map[uint32]*spdb.MigrateGVGUnitMeta)
+		chainGVGMaps := make(map[uint32]*virtualgrouptypes.GlobalVirtualGroup)
```
Comment (will-2012) [inline modular/manager/bucket_migrate_scheduler.go:72]: Refine `chainGVGMaps ` to `chainGVGMap`.

File: modular/manager/bucket_migrate_scheduler.go [hunk 2]
```diff
@@ -294,12 +294,7 @@ func (plan *BucketMigrateExecutePlan) stopSPSchedule() {
...
 
 func (plan *BucketMigrateExecutePlan) Start() error {
-	var err error
-	if err = plan.storeToDB(); err != nil {
-		log.Errorw("failed to start migrate execute plan due to store db", "error", err)
-		return err
-	}
-	log.Debugf("BucketMigrateExecutePlan Start success")
+	log.Debugf("succeed to start bucket migrate plan", "plan", plan)
 	go plan.startSPSchedule()
```
Comment (will-2012) [inline modular/manager/bucket_migrate_scheduler.go:298]: start`SP`Schedule, what's the meaning of the SP here.

File: modular/manager/bucket_migrate_scheduler.go [hunk 3]
```diff
@@ -448,29 +442,18 @@ func (s *BucketMigrateScheduler) processEvents(migrateBucketEvents *types.ListMi
...
 		if err != nil {
 			log.Errorw("failed to produce bucket migrate execute plan", "Events", migrateBucketEvents.Events, "error", err)
 			return err
 		}
-		if err = executePlan.Start(); err != nil {
-			log.Errorw("failed to start bucket migrate execute plan", "Events", migrateBucketEvents.Events, "executePlan", executePlan, "error", err)
-			return err
+		if executePlan != nil {
+			if err = executePlan.Start(); err != nil {
+				log.Errorw("failed to start bucket migrate execute plan", "Events", migrateBucketEvents.Events, "executePlan", executePlan, "error", err)
```
Comment (will-2012) [inline modular/manager/bucket_migrate_scheduler.go:452]: It is best to use lowercase letters uniformly in the log.

**Review by will-2012 — APPROVED** [review]

**Review by sysvm — APPROVED** [review]
LGTM

**Pull request merged**